### ISSUE TITLE
ARM64 Linux: jaraco.functools version 3.5.0 not available, but 3.4.0 is

### ISSUE THREAD

**Issue by sanderjo** [sabnzbd/sabnzbd#2154]
Not really problem by SABnzbd, but for SABnzbd:

Ubuntu 18.04.6 LTS, with python 3.6.9, on ARM64

Required by SABn's requirement.txt:

```
sander@nanopineo2:~/git/sabnzbd$ cat requirements.txt | grep jaraco.functools                                                                                                            

jaraco.functools==3.5.0

```

But

```
$ python3 -m pip install jaraco.functools==3.5.0 -U
Defaulting to user installation because normal site-packages is not writeable
ERROR: Could not find a version that satisfies the requirement jaraco.functools==3.5.0 (from versions: 1.0, 1.1, 1.2, 1.3, 1.4, 1.4.1, 1.5, 1.6, 1.7, 1.8, 1.8.1, 1.8.2, 1.9, 1.10, 1.11, 1.12, 1.13, 1.14, 1.15, 1.15.1, 1.15.2, 1.16, 1.17, 1.18, 1.19, 1.20, 2.0, 3.0.0, 3.0.1, 3.1.0, 3.2.0, 3.2.1, 3.3.0, 3.4.0)
ERROR: No matching distribution found for jaraco.functools==3.5.0
```

How is that possible? General problem for ARM64? Or a problem on my setup?

Ugly workaround (hoping that 3.4.0 is OK):

```
$ python3 -m pip install jaraco.functools==3.4.0 -U
Defaulting to user installation because normal site-packages is not writeable
Collecting jaraco.functools==3.4.0
  Downloading jaraco.functools-3.4.0-py3-none-any.whl (6.9 kB)
Requirement already satisfied: more-itertools in /home/sander/.local/lib/python3.6/site-packages (from jaraco.functools==3.4.0) (8.4.0)
Installing collected packages: jaraco.functools
  Attempting uninstall: jaraco.functools
    Found existing installation: jaraco.functools 3.0.1
    Uninstalling jaraco.functools-3.0.1:
      Successfully uninstalled jaraco.functools-3.0.1
Successfully installed jaraco.functools-3.4.0
```

```
$ pip3 list | grep jaraco.functools
jaraco.functools       3.4.0
```

**Comment by sanderjo** [comment]
Same for `jaraco.collections==3.5.1`:

```
ERROR: Could not find a version that satisfies the requirement jaraco.collections==3.5.1 (from versions: 1.0, 1.1, 1.2.2, 1.3, 1.3.1, 1.3.2, 1.4, 1.4.1, 1.5, 1.5.1, 1.5.2, 1.5.3, 1.6.0, 2.0, 2.1, 3.0.0, 3.1.0, 3.2.0, 3.3.0, 3.4.0)
ERROR: No matching distribution found for jaraco.collections==3.5.1
```

and  `tempora==5.0.1`

```
ERROR: Could not find a version that satisfies the requirement tempora==5.0.1 (from versions: 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.6.1, 1.7, 1.7.1, 1.8, 1.9, 1.10, 1.11, 1.12, 1.13, 1.14, 1.14.1, 2.0.0, 2.1.0, 2.1.1, 3.0.0, 4.0.0, 4.0.1, 4.0.2, 4.1.0, 4.1.1, 4.1.2)
ERROR: No matching distribution found for tempora==5.0.1
```


```
ERROR: Could not find a version that satisfies the requirement portend==3.1.0 (from versions: 0.0.0, 1.0, 1.0.1, 1.0.2, 1.1, 1.1.1, 1.1.3, 1.2, 1.3, 1.4, 1.5.1, 1.6, 1.6.1, 1.7, 1.7.1, 1.8, 2.1, 2.1.1, 2.1.2, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7.0, 2.7.1, 2.7.2, 3.0.0)
ERROR: No matching distribution found for portend==3.1.0
```

**Comment by sanderjo** [comment]
Ah, seems to be caused by python 3.6. Solved by installing and using python 3.8

```
sudo apt install python3.8 -y
python3.8 -m pip install --upgrade pip
python3.8 -m pip install -r requirements.txt -U
```

... and then running SABnzbd.py with python3.8:

```
sander@nanopineo2:~/git/sabnzbd$ python3.8 SABnzbd.py -b0 -l2
2022-04-13 22:52:54,645::INFO::[SABnzbd:1167] --------------------------------
2022-04-13 22:52:54,647::INFO::[SABnzbd:1168] SABnzbd.py-3.6.0Beta3
2022-04-13 22:52:54,918::INFO::[misc:1160] [sabnzbd.misc.run_command] Running external command: ['git', 'rev-parse', '--short', 'HEAD']
2022-04-13 22:52:54,919::DEBUG::[misc:1161] Popen arguments: {'stdin': -1, 'stdout': -1, 'stderr': -2, 'startupinfo': None, 'creationflags': 0, 'cwd': '/home/sander/git/sabnzbd'}
2022-04-13 22:52:55,093::INFO::[SABnzbd:1178] Commit = 2a25c201a
2022-04-13 22:52:55,096::INFO::[SABnzbd:1180] Full executable path = /home/sander/git/sabnzbd/SABnzbd.py
2022-04-13 22:52:55,097::INFO::[SABnzbd:1181] Arguments = "SABnzbd.py" "-b0" "-l2"
2022-04-13 22:52:55,098::INFO::[SABnzbd:1182] Python-version = 3.8.0 (default, Dec  9 2021, 17:53:27)
[GCC 8.4.0]
2022-04-13 22:52:55,099::INFO::[SABnzbd:1183] Dockerized = False
2022-04-13 22:52:55,100::INFO::[SABnzbd:1184] CPU architecture = aarch64
2022-04-13 22:52:55,233::INFO::[SABnzbd:1187] Platform = posix - Linux-5.15.25-sunxi64-aarch64-with-glibc2.27

2022-04-13 22:52:55,896::INFO::[SABnzbd:423] SABYenc module (v5.1.7)... found!
2022-04-13 22:52:55,898::INFO::[SABnzbd:424] SABYenc module is using SIMD set: NEON

```

**Comment by sanderjo** [comment]
... and let's point python3 to python3.8

`sudo ln -s  /usr/bin/python3.8 /usr/bin/python3`

**Comment by jcfp** [comment]
> Ah, seems to be caused by python 3.6. Solved by installing and using python 3.8

Jaraco bumped the minimum python version to 3.7 for all his code after 3.6 reached EOL, that's why pip doesn't find any more recent releases.
